extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/vector/DoubleVectorExpansionKernel.java

```diff
@@ -74,15 +70,12 @@ public <A extends Any> WritableChunk<A> expand(
                 offsetsDest.set(ii, result.size());
             }
             if (row != null) {
-                final DoubleConsumer consumer = result::add;
-                try (final CloseableIterator<Double> iter = row.iterator()) {
-                    Stream<Double> stream = iter.stream();
-                    if (fixedSizeLength > 0) {
-                        // limit length to fixedSizeLength
-                        stream = stream.limit(fixedSizeLength);
+                try (final CloseablePrimitiveIteratorOfDouble iter = row.iterator()) {
```

Comment:
The BarrageChunkFactoryTest doesn't check this; so we should exercise it some other way.

Reply:
Now exercised via BarrageMessageRoundTripTest; also ran into a few other issues that needed fixing.

- We need deep equals checking on ObjectVector#equals (to accommodate TableDiff validation).
- I discovered a few calls where we are not returning the provided chunk (possibly losing data when we are supposed to append).
- The columns as list header reader needs to fetch the expected size from the provided offsets; the record batch will have size 1.